### ISSUE TITLE
Intellisense fails to read <concepts> on windows.

### ISSUE THREAD

**Issue by mason-bially** [microsoft/vscode-cpptools#6818]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10.0.18363
- VS Code Version: 1.52.1
- C/C++ Extension Version: 1.2.0-insiders
- Other extensions you installed (and if the issue persists after disabling them): Bazel, C#, Rainbow Brackets, it does persist
- Does this issue involve using SSH remote to run the extension on a remote machine?: No
- The concepts file fails to be read by intellisense.

**Steps to reproduce**

add a header file with contents
```c++
#include <concepts>

class Foo { };

template<class Derived>
concept IsFooDerived =
  std::derived_from<Derived, Foo>;
```

I tried many variations of something like the following `c_cpp_properties.json`
```
{
    "configurations": [
        {
            "name": "Win32",
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE",
            ],
            "windowsSdkVersion": "10.0.17763.0",
            "cppStandard": "c++20",
            "intelliSenseMode": "msvc-x64",
            "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe",
        }
    ],
    "version": 4
}
```

**Expected behavior**
For `std::derived_from` to be found (or at the very least not cause cascading errors).

<details>
  <summary><strong>Logs</strong></summary>

```
-------- Diagnostics - 1/19/2021, 4:52:23 PM
Version: 1.2.0-insiders
Current Configuration:
{
    "name": "Win32",
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "windowsSdkVersion": "10.0.17763.0",
    "cppStandard": "c++20",
    "intelliSenseMode": "msvc-x64",
    "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe",
    "compilerArgs": [],
    "intelliSenseModeIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "compilerPathIsExplicit": true,
    "browse": {
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Data\...\repro\vscode\test.h ]:
    C:\DATA\...\REPRO\VSCODE\TEST.H
Translation Unit Configurations:
[ C:\Data\...\repro\vscode\test.h ]:
    Process ID: 15772
    Memory Usage: 12 MB
    Compiler Path: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe
    Includes:
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.28.29333\INCLUDE
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.28.29333\ATLMFC\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\CPPWINRT
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c++latest
    IntelliSense Mode: windows-msvc-x64
    Other Flags:
        --header_only_fallback
Total Memory Usage: 12 MB

------- Workspace parsing diagnostics -------
Number of folders and files enumerated: 4775
Number of files discovered (not excluded): 4750

---

File exclude: **/.git
File exclude: **/.svn
File exclude: **/.hg
File exclude: **/CVS
File exclude: **/.DS_Store
File exclude: **/.vscode
Search exclude: **/node_modules
Search exclude: **/bower_components
Search exclude: **/*.code-search
Search exclude: **/.vscode
Populate include completion cache.
Discovering files...
  Processing folder (non-recursive): C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2019/COMMUNITY/VC/TOOLS/MSVC/14.28.29333/ATLMFC/INCLUDE
  Processing folder (non-recursive): C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2019/COMMUNITY/VC/TOOLS/MSVC/14.28.29333/INCLUDE
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/CPPWINRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/SHARED/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UCRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UM/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/WINRT/
  Discovering files: 4750 file(s) processed
  0 file(s) removed from database
Done discovering files.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/Data/.../repro/vscode/test.h
Queueing IntelliSense update for files in translation unit of: C:\DATA\...\REPRO\VSCODE\TEST.H
Error squiggle count: 3
Update IntelliSense time (sec): 0.253
Code browsing service initialized
Populate include completion cache.
Discovering files...
  Processing folder (non-recursive): C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2019/COMMUNITY/VC/TOOLS/MSVC/14.28.29333/ATLMFC/INCLUDE
  Processing folder (non-recursive): C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2019/COMMUNITY/VC/TOOLS/MSVC/14.28.29333/INCLUDE
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/CPPWINRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/SHARED/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UCRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UM/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/WINRT/
  Discovering files: 4750 file(s) processed
  0 file(s) removed from database
Done discovering files.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.


```
</details>

**Screenshots**

**Additional context**

I did some of my own investigating. It appears the `<concepts>` header is guarded by:

```c++
#ifndef __cpp_lib_concepts
#pragma message("The contents of <concepts> are available only with C++20 concepts support.")
#else // ^^^ !defined(__cpp_lib_concepts) / defined(__cpp_lib_concepts) vvv
```

And `__cpp_lib_concepts` seems to be set in a previous include file `<yvals_core.h>` here:

```
#ifndef __EDG__ // TRANSITION, EDG concepts support
#define __cpp_lib_concepts 201907L
#endif // __EDG__
```

Notably `__EDG__` *does seem set* by intellisense itself. Also, even if I attempt to add `__cpp_lib_concepts` to the define list, it still fails to parse the rest of the `<concepts>` file.

**Comment by sean-mcmanus** [comment]
The issue is fixed if you update to the 14.28.29812 headers from https://visualstudio.microsoft.com/vs/preview/ and use a compilerPath like "C:/Program Files (x86)/Microsoft Visual Studio/2019/Preview/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe".

**Comment by mason-bially** [comment]
I thought I *had* updated to the latest version. Seems I had to install an entire other copy of visual studio. That's pretty frustrating.

Seems to be working now though, so thanks for the fix!
